src/MorseL.Client/Connection.cs

```diff
@@ -70,55 +103,136 @@ public Task StartAsync()
                             break;
 
                         case MessageType.ClientMethodInvocation:
-                            var invocationDescriptor = Json.DeserializeInvocationDescriptor(message.Data, _handlers);
+                            InvocationDescriptor invocationDescriptor = null;
+                            try
+                            {
+                                invocationDescriptor = Json.DeserializeInvocationDescriptor(message.Data, _handlers);
+                            }
+                            catch (Exception exception)
+                            {
+                                await HandleUnparseableReceivedInvocationDescriptor(message.Data, exception).ConfigureAwait(false);
+                                return;
+                            }
+
                             if (invocationDescriptor == null)
                             {
-                                _logger?.LogDebug("Invocation request for unknown hub method");
+                                // Valid JSON but unparseable into a known, typed invocation descriptor (unknown method name, invalid parameters)
```

Comment:
Go for it. Truthfully the preview package is my only real concern but I'm okay with that as well. :ship: